src/qml/filters/fisheye/ui.qml

```diff
@@ -0,0 +1,655 @@
+            i += 1
+        } while(v_more)
+    }
+
+    //lens presets
+    function lensValue(index)  { return floatFromIndex(index, lensCombo.count) }
+    //scale presets
+    function scaleValue(index)   { return floatFromIndex(index, scaleCombo.count) }
+    //quality presets
+    function qualityValue(index)  { return floatFromIndex(index, qualityCombo.count) }
+    //aspect ratio presets
+    function aspectValue(index)    { return floatFromIndex(index, aspectCombo.count) }
+
+    //calculate a float for plugin. 1-based
+    function floatFromIndex(idx, max) {
+        if (idx < 1)
+            idx = 1
+        if (idx > max)
+            idx = max
+        return ((1 / (max-1)) * (idx-1)).toPrecision(6)
+    }
+
+    //convert float to int for UI inputs. 1-based
+    function indexFromFloat(idx, max) {
+        var v_rng = 1 / max
+        var v_plug = filter.getDouble(idx)
+        for (var i = 1; i < max; i++) {
+            if (v_plug < (v_rng * i)) {
+                return i
+            }
+        }
+        return max
+    }
+
+    //index for reset button. 1-based
+    function setDefaultIndex(value, max){
+        var v_rng = 1 / max
+        for (var i = 1; i < max; i++) {
+            if (value < (v_rng * i))
+                return i
+        }
+        return max
+    }
+
+    //set presets
+    function setPresetData() {
+        var v_more = true
+        var v_ret = ''
+        var i = 0
+
+        do {
+            v_ret = getPresetData(i)
+            if (v_ret !== '' ){
+                setPluginData(v_ret[0], v_ret[1], v_ret[2], v_ret[3], v_ret[4], v_ret[5], v_ret[6], v_ret[7])
+                filter.savePreset(preset.parameters, qsTr(v_ret[8]+ ' ' +v_ret[9] + ' (' +v_ret[10]+')'))
+            } else {
+                v_more = false
+            }
+            i += 1
+        } while(v_more)    
+    }
+
+    Component.onCompleted: {
+        filter.blockSignals = true
+        if (filter.isNew) {
+            setPresetData()
+            
+            // save (default) in preset dropdown
+            setPluginData(focalRatioDefault, deFishDefault, lensTypeDefault, scalePresetDefault, scaleManualDefault, qualityDefault, aspectPresetDefault, aspectManualDefault)
+            filter.savePreset(defaultParameters)
+        }
+        filter.blockSignals = false
+        
+        if (cameraShowNew) fillCameraCombo()
+
+        setControlData()
+    }
+
+    GridLayout {
+        columns: 3
+        anchors.fill: parent
+        anchors.margins: 8
+
+        // Row 1 Preset
+        Label {
+            text: qsTr('Presets')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.Preset {
+            id: preset
+            parameters: defaultParameters
+            onPresetSelected: setControlData()
+        }
+        Row {width: 6} //blank
+
+        //line seperator #1
+        Rectangle{Layout.columnSpan: 3; Layout.fillWidth: true; height: 1; color: activePalette.text; opacity: 0.3}

```

Comment:
Remove this horizontal line. There are way too many lines in this UI. Lines are visual distraction. I allow some below because there is preset vs. manual so those rows are grouped.